Sources/Schema.swift

```diff
@@ -2,7 +2,7 @@ import ObjectMapper
 
 /// Schemas are used to define the types used in body parameters. They are more expressive than Items.
 public enum Schema {
-    indirect case structure(Structure<Schema>)
+    indirect case structure(SchemaStructure)
```

Comment:
Instead of creating a new type for this, how about we just add the metadata to this case?
I.e. ```case structure(metadata: Metadata, structure: Structure<Schema>)```